src/components/LoginModal.tsx

```diff
@@ -0,0 +1,170 @@
+import { Box, Button, Checkbox, FormControlLabel, FormGroup, Modal, Typography } from "@mui/material";
+import AccountCircleIcon from "@mui/icons-material/AccountCircle";
+import LockIcon from '@mui/icons-material/Lock';
+import theme from "@/app/theme";
+import IconInput from "@/components/IconInput";
+import { useState } from "react";
+
+
+
+
+
+interface LoginProps {
+  isOpen: boolean;
+  login: (username:string, password:string)=>Promise<string>;
+  signup: (username:string, password:string, confirm:string)=>Promise<string>;
+}
+const LoginModal = ({isOpen, login, signup}:LoginProps) => {
+
+  const [isSignup, setIsSignup] = useState(false);
+
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [checkedReminder, setReminderChecked] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+
+  const handleSubmit = ()=>{
+    if(isSignup){
+      signup(username,password,confirmPassword).then(message=>{
+        setErrorMessage(message)
+      })
+    }else{
+      login(username,password).then(message=>{
+        setErrorMessage(message);
+      })
+    }
+  }
+
+  return (
+    <Modal
+      open={isOpen}
+    >
+      <Box sx={{
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        transform: 'translate(-50%, -50%)',
+        width: 400,
+        height: 500,
+        background: theme.palette.secondary.main,
+        border: '2px solid #000',
+        outline:"none",
+        boxShadow: 24,
+        p: 4,
+        borderRadius: 2,
+        display: 'flex',
+        flexDirection: 'column',
+
+      }}>
+        <Typography data-testid={"modal-title"} id="modal-title" variant="h3" component="h2">
```

Comment:
If possible, let's keep the same spacing between the title and the form in Login and in Sign up. It doesn't matter if the modal is shorter in one.